core/server/models/plugins/filter.js

```diff
@@ -191,36 +193,31 @@ filter = function filter(Bookshelf) {
         },
 
         /**
-         * ## Apply Filters
-         * Method which makes the necessary query builder calls (through knex) for the filters set
-         * on this model instance
-         * @param {Object} options
-         * @returns {Bookshelf.Model}
+         * Method which makes the necessary query builder calls (through knex) for the filters set on this model
+         * instance.
          */
         applyDefaultAndCustomFilters: function applyDefaultAndCustomFilters(options) {
-            var self = this;
-
-            // @TODO figure out a better place/way to trigger loading filters
-            if (!this._filters) {
-                this.fetchAndCombineFilters(options);
-            }
-
-            if (this._filters) {
-                if (this.debug) {
-                    gql.json.printStatements(this._filters.statements);
-                }
-
-                this.preProcessFilters(options);
-
-                this.query(function (qb) {
-                    gql.knexify(qb, self._filters);
+            const nql = require('@nexes/nql');
+
+            const customFilter = options.filter;
+            const defaultFilters = this.enforcedFilters(options);
+            const enforcedFilters = this.enforcedFilters(options);
+
+            debug('custom', customFilter);
+            debug('default', defaultFilters);
+            debug('enforced', enforcedFilters);
+
+            if (customFilter) {
+                this.query((qb) => {
+                    nql(customFilter, {relations: {tags: {
```

Comment:
We will sort this out later. Can stay like this for now. No priority, as long as it works.